tensorcircuit/templates/lattice.py

```diff
@@ -759,20 +855,20 @@ class SquareLattice(TILattice):
     def __init__(
         self,
         size: Tuple[int, int],
-        lattice_constant: float = 1.0,
+        lattice_constant: Union[float, Any] = 1.0,
         pbc: Union[bool, Tuple[bool, bool]] = True,
         precompute_neighbors: Optional[int] = None,
     ):
         """Initializes the SquareLattice."""
         dimensionality = 2
 
-        # Define lattice vectors for a square lattice
-        lattice_vectors = np.array([[lattice_constant, 0.0], [0.0, lattice_constant]])
-
-        # A square lattice has a single site in its basis
-        basis_coords = np.array([[0.0, 0.0]])
+        # Define orthogonal lattice vectors for a square.
+        lattice_vectors = backend.convert_to_tensor(
+            [[lattice_constant, 0.0], [0.0, lattice_constant]]
```

Comment:
if lattice constant is already a tensor, is this form of definition for lattice vectors work?